modules/integration/tests-integration/src/test/java/org/wso2/jaggery/integration/tests/hostObjects/core/FileHostObjectTestCase.java

```diff
@@ -35,139 +36,138 @@
  */
 public class FileHostObjectTestCase {
 
-    @Test(groups = {"jaggery"},
-          description = "Test for file host object")
-    public void testFileExist() {
+    private static final Log log = LogFactory.getLog(FileHostObjectTestCase.class);
+
+    @Test(groups = { "jaggery" }, description = "Test for file host object")
+    public void testFileExist() throws IOException {
+        ClientConnectionUtil.waitForPort(9763);
+        String finalOutput = null;
+        BufferedReader in = null;
+        try {
+            URL jaggeryURL = new URL("http://localhost:9763/testapp/file.jag");
+            URLConnection jaggeryServerConnection = jaggeryURL.openConnection();
+            in = new BufferedReader(new InputStreamReader(jaggeryServerConnection.getInputStream()));
+            String inputLine;
+            while ((inputLine = in.readLine()) != null) {
+                finalOutput = inputLine;
+            }
+            in.close();
+        } catch (IOException e) {
+            log.error(e);
+        } finally {
+            in.close();
```

Comment:
It's a good idea to to check in == null before trying to do any operation. Please change that.